Anytype/Sources/Models/Documents/Document/BaseDocument.swift

```diff
@@ -214,19 +214,21 @@ final class BaseDocument: BaseDocumentProtocol {
         }
     }
     
-    private func setupView(_ model: ObjectViewModel) async {
+    @MainActor
+    private func setupView(_ model: ObjectViewModel) {
         viewModelSetter.objectViewUpdate(model)
         isOpened = true
-        await setupSubscriptions()
+        setupSubscriptions()
         triggerSync(updates: [.general])
     }
     
-    private func setupSubscriptions() async {
+    @MainActor
+    private func setupSubscriptions() {
         syncStatusStorage.statusPublisher(spaceId: spaceId).sink { [weak self] syncStatus in
             self?.syncStatusSubject.send(syncStatus)
         }.store(in: &subscriptions)
         
-        await accountParticipantsStorage.canEditPublisher(spaceId: spaceId).sink { [weak self] canEdit in
+        accountParticipantsStorage.canEditPublisher(spaceId: spaceId).sink { [weak self] canEdit in
```

Comment:
Theory. I couldn't reproduce it.
This `await` call changed thread and make a race for `subscriptions` array.

Reply:
`.store(in: &subscriptions)` 235 line can be call in different thread

Reply:
Base document should be actor or mainactor)